fluid-common/scigen.fld

```diff
@@ -15,7 +15,35 @@ let rankLabel word n =
    if n == 1 then word
    else ordinal n ++ "-" ++ word;
 
-let trendWord n1 n2 w1 w2 w3 =
-    if n1 > n2 then w1
-    else if n1 < n2 then w2
-    else w3;
+let trendWord n1 n2 compareWord =
+    compareWord (compare n1 n2);
+
+let growShrink EQ = "unchanging";
+    growShrink LT = "shrinking";
+    growShrink GT = "growing";
+
+let smallerHigher EQ = "equal";
+    smallerHigher LT = "smaller";
+    smallerHigher GT = "larger";
+
+let shorterLonger EQ = "equal";
+    shorterLonger LT = "shorter";
+    shorterLonger GT = "longer";
+
+let betterWorse EQ = "equal";
+    betterWorse LT = "worse";
+    betterWorse GT = "better";
+
+let improveWorsed EQ = "equal performance to that of";
+    improveWorsed LT = "underperforming";
+    improveWorsed GT = "outperforming";
+
+-- (a -> Number) -> List a -> a
+let maximumBy f lst =
```

Comment:
I think this can probably be expressed directly (without `helper`), or alternatively using `foldr1` (see `maximum` and `minimum` in prelude).